steampipecloud/resource_steampipecloud_connection.go

```diff
@@ -483,138 +301,55 @@ func resourceConnectionDelete(ctx context.Context, d *schema.ResourceData, meta
 	return diags
 }
 
-func CreateConnectionConfiguration(d *schema.ResourceData) (ConnectionConfig, error) {
-	var connConfig ConnectionConfig
-
-	if value, ok := d.GetOk("access_key"); ok {
-		connConfig.AccessKey = value.(string)
-	}
-	if value, ok := d.GetOk("api_key"); ok {
-		connConfig.ApiKey = value.(string)
-	}
-	if value, ok := d.GetOk("base_url"); ok {
-		connConfig.BaseURL = value.(string)
-	}
-	if value, ok := d.GetOk("bearer_token"); ok {
-		connConfig.BearerToken = value.(string)
-	}
-	if value, ok := d.GetOk("client_id"); ok {
-		connConfig.ClientID = value.(string)
-	}
-	if value, ok := d.GetOk("client_secret"); ok {
-		connConfig.ClientSecret = value.(string)
-	}
-	if value, ok := d.GetOk("credentials"); ok {
-		creds := value.(string)
-		buffer := new(bytes.Buffer)
-		if err := json.Compact(buffer, []byte(creds)); err != nil {
-			log.Println(err)
-		}
-		connConfig.Credentials = buffer.String()
-	}
-	if value, ok := d.GetOk("database_id"); ok {
-		connConfig.DatabaseID = value.(string)
+// config is a json string
+// apply standard formatting to old and new data then compare
+func suppressIfDataMatches(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
```suggestion
func connectionJSONStringsEqual(k, old, new string, d *schema.ResourceData) bool {
```

I know this function is used as the `DiffSuppressFunc` function, but difficult to tell what it's doing just by name